File: clinic-admin/src/api/operationLog.ts
import { get, post } from '@/utils/request'

export interface OperationLog {
  id: number
  clinicId: number
  userId: number
  username: string
  realName: string
  module: string
  operationType: number
  operationTypeName?: string
  description: string
  requestMethod: string
  requestUrl: string
  requestParams?: string
  responseData?: string
  ipAddress: string
  location?: string
  userAgent?: string
  status: number
  errorMsg?: string
  executionTime?: number
  operationTime: string
  createdAt: string
}

export interface OperationLogQuery {
  clinicId: number
  module?: string
  operationType?: number
  username?: string
  startTime?: string
  endTime?: string
  status?: number
  pageNum?: number
  pageSize?: number
}

// 操作类型选项
export const operationTypeOptions = [
  { label: '新增', value: 1 },
  { label: '修改', value: 2 },
  { label: '删除', value: 3 },
  { label: '查询', value: 4 },
  { label: '导出', value: 5 },
  { label: '登录', value: 6 },
  { label: '登出', value: 7 },
  { label: '其他', value: 8 },
  { label: '审核', value: 9 },
  { label: '导入', value: 10 },
  { label: '打印', value: 11 },
  { label: '支付', value: 12 },
  { label: '退费', value: 13 }
]

// 获取操作类型名称
export const getOperationTypeName = (type: number) => {
  const item = operationTypeOptions.find(o => o.value === type)
  return item?.label || '其他'
}

// 分页查询操作日志
export const getOperationLogList = (params: OperationLogQuery) =>
  get<{ total: number; list: OperationLog[] }>('/admin/operation-log/list', params)

// 批量删除日志
export const deleteOperationLogs = (ids: number[]) =>
  post('/admin/operation-log/delete', ids)

// 清理日志
export const cleanOperationLogs = (days: number) =>
  post(`/admin/operation-log/clean?days=${days}`)

File: clinic-api/src/main/java/com/clinic/entity/OperationLog.java
package com.clinic.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 操作日志审计表
 */
@Data
@TableName("operation_log")
public class OperationLog implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 诊所ID
     */
    private Long clinicId;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 用户名
     */
    private String username;

    /**
     * 用户真实姓名
     */
    private String realName;

    /**
     * 操作模块
     */
    private String module;

    /**
     * 操作类型：1新增 2修改 3删除 4查询 5导出 6登录 7登出 8其他
     */
    private Integer operationType;

    /**
     * 操作描述
     */
    private String description;

    /**
     * 请求方法
     */
    private String requestMethod;

    /**
     * 请求URL
     */
    private String requestUrl;

    /**
     * 请求参数
     */
    @TableField(typeHandler = com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler.class)
    private String requestParams;

    /**
     * 响应结果
     */
    @TableField(typeHandler = com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler.class)
    private String responseData;

    /**
     * IP地址
     */
    private String ipAddress;

    /**
     * 地理位置
     */
    private String location;

    /**
     * 浏览器UA
     */
    private String userAgent;

    /**
     * 操作状态：0失败 1成功
     */
    private Integer status;

    /**
     * 错误信息
     */
    private String errorMsg;

    /**
     * 执行时长（毫秒）
     */
    private Long executionTime;

    /**
     * 操作时间
     */
    private LocalDateTime operationTime;

    /**
     * 创建时间
     */
    //@TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
